src/app/core/models/part.ts

```diff
@@ -16,4 +16,5 @@ export interface Part {
   readonly isPlanned: boolean
   readonly isUntimed: boolean
   readonly metadata?: unknown
+  readonly replacedPieces: Piece[]
```

Comment:
I'm hesitant to call it `replacedPlannedPieces`. Mainly because right now it is `part-entity.service` that only puts Planned Pieces in it. But `part-entity.service` is a different file. So if we call the attribute `replacedPlannedPieces` then we are naming an attribute in one file after how _another file_ uses it. 

There is nothing in this interface that can force it to be planned Pieces, so it's entirely possible (and legal) for some other file to put an unplanned Piece in this array.

Now, if this was a class that had full control of what goes into the array then I would be all for it to call it `replacedPlannedPieces`, but since we have taken the functional approach here we don't have that same guarentee.

Reply:
Changing the type for this to only contain planned pieces is simple (either by extension or type intersection), in which case it is guaranteed. If it is allowed to put an unplanned piece in this, then the code prior to [this change](https://github.com/tv2/ng-sofie/pull/204#discussion_r1604424074) would have had a hidden dependency.

I am fine with keeping it as is.